src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -1023,141 +1041,179 @@ private void OutputMethodNode(CppMethodCodeNode methodCodeNode)
                 Out.Write(sb.ToString());
             }
         }
-
         private void OutputTypeNode(IEETypeNode typeNode, NodeFactory factory, CppGenerationBuffer forwardDefinitions, CppGenerationBuffer typeDefinitions, CppGenerationBuffer methodTable)
         {
             if (_emittedTypes == null)
             {
                 _emittedTypes = new HashSet<TypeDesc>();
             }
+            if (!_emittedTypes.Contains(typeNode.Type))
```

Comment:
This check is redundant - done again a few lines later. (Also, it would look better to have the early out as before so that most of the function does not need to be indented.)